extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/PythonTableDataService.java

```diff
@@ -190,15 +223,23 @@ public SafeCloseable subscribeToTableLocations(
                 @NotNull final TableKeyImpl tableKey,
                 @NotNull final Consumer<TableLocationKeyImpl> tableLocationListener,
                 @NotNull final Runnable successCallback,
-                @NotNull final Consumer<String> failureCallback) {
+                @NotNull final Consumer<RuntimeException> failureCallback) {
             final BiConsumer<TableLocationKeyImpl, ByteBuffer[]> convertingListener =
-                    (tableLocationKey, byteBuffers) -> processTableLocationKey(definition, tableKey,
-                            tableLocationListener,
-                            tableLocationKey, byteBuffers);
+                    (tableLocationKey, byteBuffers) -> {
+                        try {
+                            processTableLocationKey(
+                                    definition, tableKey, tableLocationListener, tableLocationKey, byteBuffers);
+                        } catch (final RuntimeException e) {
+                            failureCallback.accept(e);
```

Comment:
This may not be ideal. You'll unblock the caller and declare the subscription dead, but you won't have canceled the underlying subscription properly. You may need to distinguish between the two failure modes (underlying failure and conversion failure) and cancel the sub automatically in the latter case.